### DISCUSSION TITLE
How can I achieve text shadow in Avalonia?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#12649] [Q&A] [answered]
I want to achieve text shadow in Avalonia. According to the text shadow tutorial of WPF:
https://learn.microsoft.com/en-us/dotnet/desktop/wpf/advanced/how-to-create-text-with-a-shadow?view=netframeworkdesktop-4.8

```
<TextBlock
  Text="Shadow Text"
  Foreground="Teal">
  <TextBlock.Effect>
    <DropShadowEffect
      ShadowDepth="4"
      Direction="330"
      Color="Black"
      Opacity="0.5"
     BlurRadius="4"/>
  </TextBlock.Effect>
</TextBlock>
```

It doesn't work in Avalonia any. Moreover, I cannot find any Avalonia tutorial about text shadow on google.

How can I achieve this?

**Comment by kekekeks** [comment, marked as answer]
The counterpart of WPF's `DropShadowEffect` is `DropShadowDirectionEffect`.

**Reply by kyuranger** [reply]
It works now! Thank you.
